testing/endtoend/evaluators/fee_recipient.go

```diff
@@ -0,0 +1,106 @@
+package evaluators
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/common/hexutil"
+	"github.com/ethereum/go-ethereum/ethclient"
+	"github.com/ethereum/go-ethereum/rpc"
+	"github.com/pkg/errors"
+	fieldparams "github.com/prysmaticlabs/prysm/config/fieldparams"
+	"github.com/prysmaticlabs/prysm/config/params"
+	ethpb "github.com/prysmaticlabs/prysm/proto/prysm/v1alpha1"
+	"github.com/prysmaticlabs/prysm/testing/endtoend/components"
+	"github.com/prysmaticlabs/prysm/testing/endtoend/helpers"
+	e2e "github.com/prysmaticlabs/prysm/testing/endtoend/params"
+	"github.com/prysmaticlabs/prysm/testing/endtoend/policies"
+	"github.com/prysmaticlabs/prysm/testing/endtoend/types"
+	log "github.com/sirupsen/logrus"
+	"google.golang.org/grpc"
+	"google.golang.org/protobuf/types/known/emptypb"
+)
+
+var FeeRecipientIsPresent = types.Evaluator{
+	Name:       "Fee_Recipient_Is_Present_%d",
+	Policy:     policies.AfterNthEpoch(helpers.BellatrixE2EForkEpoch),
+	Evaluation: feeRecipientIsPresent,
+}
+
+func feeRecipientIsPresent(conns ...*grpc.ClientConn) error {
+	conn := conns[0]
+	client := ethpb.NewBeaconChainClient(conn)
+	chainHead, err := client.GetChainHead(context.Background(), &emptypb.Empty{})
+	if err != nil {
+		return errors.Wrap(err, "failed to get chain head")
+	}
+	req := &ethpb.ListBlocksRequest{QueryFilter: &ethpb.ListBlocksRequest_Epoch{Epoch: chainHead.HeadEpoch.Sub(1)}}
+	blks, err := client.ListBeaconBlocks(context.Background(), req)
+	if err != nil {
+		return errors.Wrap(err, "failed to list blocks")
+	}
+
+	rpcclient, err := rpc.DialHTTP(fmt.Sprintf("http://127.0.0.1:%d", e2e.TestParams.Ports.Eth1RPCPort))
+	if err != nil {
+		return err
+	}
+	defer rpcclient.Close()
+	web3 := ethclient.NewClient(rpcclient)
+	ctx := context.Background()
+
+	for _, ctr := range blks.BlockContainers {
+		switch ctr.Block.(type) {
+		case *ethpb.BeaconBlockContainer_BellatrixBlock:
+			var account common.Address
+
+			fr := ctr.GetBellatrixBlock().Block.Body.ExecutionPayload.FeeRecipient
+			if len(fr) != 0 && hexutil.Encode(fr) != params.BeaconConfig().EthBurnAddressHex {
+				account = common.BytesToAddress(fr)
+			} else {
+				return errors.New("fee recipient is not set")
+			}
+			validatorRequest := &ethpb.GetValidatorRequest{
+				QueryFilter: &ethpb.GetValidatorRequest_Index{
+					Index: ctr.GetBellatrixBlock().Block.ProposerIndex,
+				},
+			}
+			validator, err := client.GetValidator(context.Background(), validatorRequest)
+			if err != nil {
+				return errors.Wrap(err, "failed to get validators")
+			}
+			publickey := validator.GetPublicKey()
+			// calculate deterministic fee recipient using first 20 bytes of public key
+			deterministicFeeRecipient := common.HexToAddress(hexutil.Encode(publickey[:fieldparams.FeeRecipientLength])).Hex()
+			if deterministicFeeRecipient != account.Hex() && components.DefaultFeeRecipientAddress != account.Hex() {
```

Comment:
As a last point, is there anyway to differentiate which validator should be running with the default recipients and which validators should not be ? It is possible (ex:some validator bug) , where the validator does not apply the proposer config correctly and just uses the default fee recipient. With this check we wouldn't have anyway to differentiate. If this isn't possible to figure out currently it is fine we can ignore it for now.